.travis.yml

```diff
@@ -1,38 +1,47 @@
-language: python
+language: c
 
-python:
-    - 2.7
+os:
+    - osx
+    - linux
 
-env: DISPLAY=:99 MOZ_XVFB=1
+compiler:
+    - gcc
 
 before_install:
-    - sudo apt-get update
-    - "/sbin/start-stop-daemon --start --quiet --make-pidfile --pidfile /tmp/custom_xvfb_99.pid --background --exec /usr/bin/Xvfb -- :99 -ac -screen 0 1024x768x24"
+    # set up linux only items
+    - ./.travis/preinstall.sh
 
-install:
-    - sudo apt-get install flashplugin-nonfree subversion
+    - sudo -H pip install virtualenv
+    - virtualenv ~/.venv
+    - source ~/.venv/bin/activate
```

Comment:
It's cleanest to put the lines creating and activating the virtualenv at the top level, in the .travis..yml. They're cross-platform.

I've moved the virtualenv install, needed for linux only, to the before_install.sh script, where I'm using apt-get instead of pip.